jupyterlab/extensions/pypi.py

```diff
@@ -0,0 +1,408 @@
+"""Extensions manager using pip as package manager and PyPi.org as packages source."""
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import asyncio
+import io
+import json
+import re
+import sys
+import xmlrpc.client
+from datetime import datetime, timedelta
+from functools import partial
+from itertools import groupby
+from pathlib import Path
+from subprocess import CalledProcessError, run
+from tarfile import TarFile
+from typing import Any, Callable, Dict, List, Optional, Tuple
+from zipfile import ZipFile
+
+import tornado
+from async_lru import alru_cache
+from traitlets import CFloat, Unicode, config
+
+from jupyterlab.extensions.manager import (
+    ActionResult,
+    ExtensionPackage,
+    ExtensionsManager,
+)
+
+
+@alru_cache(maxsize=500)
+async def _fetch_package_metadata(name: str, latest_version: str, base_url: str) -> dict:
+    http_client = tornado.httpclient.AsyncHTTPClient()
+    response = await http_client.fetch(
+        base_url + f"/{name}/{latest_version}/json",
+        headers={"Content-Type": "application/json"},
+    )
+    data = json.loads(response.body).get("info")
+
+    # Keep minimal information to limit cache size
+    return {k: data.get(k) for k in ["summary", "home_page", "package_url", "author", "license"]}
+
+
+class PyPiExtensionsManager(ExtensionsManager):
```

Comment:
Maybe this could be renamed to the following for consistency with "PyPI" and the fact we usually refer to the "extension manager":

```suggestion
class PyPIExtensionManager(ExtensionsManager):
```